### PR TITLE
fix: makefile improvence - RK-19036

### PR DESCRIPTION
**Pull request by gosharo**
- fix: add e2e test
- fix: add e2e test
- fix: add e2e test
- fix: makefile improvence

### Description

Please provide a brief description of the changes made in this pull request.

### Related Issue(s)

If this pull request addresses or relates to any open issues, please mention them here using the syntax `Fixes #<issue_number>` or `Resolves #<issue_number>`.

### Checklist

Before submitting this pull request, please ensure that you have completed the following tasks:

- [ ] Reviewed the [Contributing Guidelines](../docs/CONTRIBUTING.md) for the Piper project.
- [ ] Ensured that your changes follow the [coding guidelines and style](../docs/CONTRIBUTING.md#coding-guidelines) of the project.
- [ ] Run the tests locally and made sure they pass.
- [ ] Updated the relevant documentation, if applicable.

### Testing Instructions

Please provide clear instructions on how to test and verify the changes made in this pull request.

### Additional Information

Add any additional information or context that would be helpful in understanding and reviewing this pull request.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -39,23 +39,26 @@ jobs:
...
       - uses: docker/setup-buildx-action@v2
-      - name: Build Docker Image
-        uses: docker/build-push-action@v4
-        with:
-          context: .
-          push: false
-          tags: localhost:5001/piper:latest
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
 #      - name: install kind
```
Comment (qadiludmer) [inline .github/workflows/ci.yaml:42]: why comment ?

File: makefile [hunk 2]
```diff
@@ -10,13 +10,40 @@ local-build:
 
 .PHONY: init-kind
 init-kind:
-	@if [[ "$(kind get clusters -q | grep piper)" == "" ]]; then sh ./scripts/init-kind.sh; else echo "Kind piper exists, switching context"; fi
+ifeq ($(kind get clusters -q | grep piper), "")
+	sh ./scripts/init-kind.sh
+else
```
Comment (qadiludmer) [inline makefile:15]: else not needed in this case cause you switch context either way

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (gosharo) [reply .github/workflows/ci.yaml:42]: working n it in different branch... saving energy to the planet

File: makefile (hunk 2 above)
Comment (gosharo) [reply makefile:15]: it's needed because the script doing shit-load of configuration.

File: scripts/init-kind.sh [hunk 3]
```diff
@@ -74,14 +74,4 @@ data:
...
-kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/main/deploy/static/provider/kind/deploy.yaml && \
-kubectl wait --namespace ingress-nginx \
-       --for=condition=ready pod \
-       --selector=app.kubernetes.io/component=controller \
-       --timeout=90s
-
-# 7. Install argo workflows
-helm upgrade --install argo-workflow argo/argo-workflows -n workflows --create-namespace -f workflows.values.yaml
\ No newline at end of file
+sleep 30
```
Comment (qadiludmer) [inline scripts/init-kind.sh:77]: why sleep ?

Comment (qadiludmer) [reply scripts/init-kind.sh:77]: can't do something like 

```
kubectl wait --namespace workflows\
       --for=condition=ready pod \
       --selector=argo-whatever ?\
       --timeout=180s
```

File: makefile (hunk 2 above)
Comment (qadiludmer) [reply makefile:15]: so the echo message should be just "skipping cluster installation"

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (qadiludmer) [reply .github/workflows/ci.yaml:42]: you don't need to comment. git history take care for it

File: scripts/init-kind.sh (hunk 3 above)
Comment (gosharo) [reply scripts/init-kind.sh]: fixed

File: makefile (hunk 2 above)
Comment (gosharo) [reply makefile:15]: fixed

**Review by qadiludmer — APPROVED** [review]

**Pull request merged**